pkg/ceph/osd/device.go

```diff
@@ -611,15 +617,10 @@ func addOSDToCrushMap(context *clusterd.Context, config *osdConfig, clusterName,
 	weight, _ = strconv.ParseFloat(fmt.Sprintf("%.4f", weight), 64)
 
 	osdEntity := fmt.Sprintf("osd.%d", osdID)
-	locArgs, err := client.FormatLocation(location)
-	if err != nil {
-		return err
-	}
-
 	logger.Infof("adding %s (%s), bytes: %d, weight: %.4f, to crush map at '%s'",
-		osdEntity, osdDataPath, totalBytes, weight, strings.Join(locArgs, " "))
+		osdEntity, osdDataPath, totalBytes, weight, location)
 	args := []string{"osd", "crush", "create-or-move", strconv.Itoa(osdID), fmt.Sprintf("%.4f", weight)}
-	args = append(args, locArgs...)
+	args = append(args, strings.Split(location, " ")...)
```

Comment:
are spaces in any of the CRUSH location fields supported at the ceph level?  the way we are handling them it looks like Rook would not support any spaces.

Reply:
while i can't find anything in the documentation about spaces in the crush types, it seems prone to disaster. The `crush location` in the config file is space-delimited as well and I haven't seen any examples putting quotes around the names.